### PR TITLE
Only do gracefully stop when DN Stop

### PR DESCRIPTION
**Pull request by XuanYang-cn**
/kind improvement

### CODE REVIEW

**Comment by XuanYang-cn** [comment]
rerun ut

File: internal/util/flowgraph/flow_graph.go [hunk 1]
```diff
@@ -108,8 +118,9 @@ func (fg *TimeTickedFlowGraph) Close() {
 // NewTimeTickedFlowGraph create timetick flowgraph
 func NewTimeTickedFlowGraph(ctx context.Context) *TimeTickedFlowGraph {
 	flowGraph := TimeTickedFlowGraph{
-		nodeCtx: make(map[string]*nodeCtx),
-		closeWg: &sync.WaitGroup{},
+		nodeCtx:         make(map[string]*nodeCtx),
+		closeWg:         &sync.WaitGroup{},
+		closeGracefully: atomic.NewBool(CloseImmediately),
```
Comment (czs007) [inline internal/util/flowgraph/flow_graph.go:123]: default is CloseImmediately, why ?

File: internal/util/flowgraph/input_node.go [hunk 2]
```diff
@@ -60,14 +68,27 @@ func (inNode *InputNode) Name() string {
...
-			log.Info("trigger force sync",
-				zap.Int64("collection", inNode.collectionID),
+		log := log.With(
+			zap.String("node", inNode.Name()),
+			zap.Int64("collection", inNode.collectionID),
+		)
+		log.Info("input node message stream closed",
+			zap.Bool("closeGracefully", inNode.closeGracefully.Load()),
+		)
+		if inNode.lastMsg != nil && inNode.closeGracefully.Load() {
```
Comment (czs007) [inline internal/util/flowgraph/input_node.go:90]: As default is CloseImmediately, this pr will change original behavior.

File: internal/util/flowgraph/flow_graph.go (hunk 1 above)
Comment (XuanYang-cn) [reply internal/util/flowgraph/flow_graph.go:123]: Only need to close gracefully when doing rolling upgrades, This is what the PR for. Gracefully will try to clean all buffered msgs in the flowgraph and update meta, we don't want to do it every time when we release a flowgraph, for example, when dropping collections, when balance channels.

**Pull request merged**